### ISSUE TITLE
[Question] Do steps should fail with soft expect?

### ISSUE THREAD

**Issue by vincenzo-gasparo** [microsoft/playwright#16904]
As a Playwright user,
I would like to know if a failed expect.soft assertion should change the test step status

Example snippet:

```
import test, { expect } from "@playwright/test";

test("Example Test", async () => {
  await test.step("Step that succeeds", async () => {
    expect.soft(2 + 2).toEqual(4);
  });

  await test.step("Step that fails", async () => {
    expect.soft(2 + 2).toEqual(5);
  });

  await test.step("Another step that succeeds", async () => {
    expect.soft(2 + 2).toEqual(4);
  });
});
```

As you can see the test report shows the test step as passed, is it the right behavior?

<img width="991" alt="image" src="https://user-images.githubusercontent.com/10702614/187307227-dc7ed0b4-20fe-4d38-9ab4-3c903f25db33.png">

Thanks

**Comment by pavelfeldman** [comment]
Soft asserts only fail in the end of the entire test. Would you want to assert them in the end of your step manually?

**Comment by vincenzo-gasparo** [comment]
It's right to let assertions fail at the end of the test but it would make sense to mark test steps with failed status since failing soft assertions  will cause a test fail. In my opinion the report would be more clear in this way

**Comment by odinho** [comment]
This would be quite useful. But also having it mention what step this failed on in the report, as well as showing it in the console output while running.

**Comment by dgozman** [comment]
This was actually implemented in #19973.
